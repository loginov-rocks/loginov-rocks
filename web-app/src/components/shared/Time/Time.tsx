import * as React from 'react';

interface Props {
  label?: string;
  minutes: number;
}

export const Time: React.FC<Props> = ({ label, minutes }) => {
  const hours = Math.floor(minutes / 60);
  const remainingMinutes = minutes % 60;

  let string = '';

  if (hours > 0) {
    string += `${hours} hour${hours !== 1 ? 's' : ''}`;
  }

  if (remainingMinutes > 0) {
    if (string.length > 0) {
      string += ', ';
    }

    string += `${remainingMinutes} minute${remainingMinutes !== 1 ? 's' : ''}`;
  }

  if (string === '') {
    return null;
  }

  return <span title={label}>{string}</span>;
};

Time.defaultProps = {
  label: undefined,
};
