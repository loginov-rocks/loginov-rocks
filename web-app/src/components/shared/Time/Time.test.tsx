/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import { Time } from './Time';

it('matches snapshot when 0 minutes passed', () => {
  const wrapper = shallow(<Time label="Time to Complete" minutes={0} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when 1 minute passed', () => {
  const wrapper = shallow(<Time label="Time to Complete" minutes={1} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when 12 minutes passed', () => {
  const wrapper = shallow(<Time label="Time to Complete" minutes={12} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when 60 minutes passed', () => {
  const wrapper = shallow(<Time label="Time to Complete" minutes={60} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when 120 minutes passed', () => {
  const wrapper = shallow(<Time label="Time to Complete" minutes={120} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when 121 minutes passed', () => {
  const wrapper = shallow(<Time label="Time to Complete" minutes={121} />);

  expect(wrapper).toMatchSnapshot();
});

it('matches snapshot when 123 minutes passed', () => {
  const wrapper = shallow(<Time label="Time to Complete" minutes={123} />);

  expect(wrapper).toMatchSnapshot();
});
