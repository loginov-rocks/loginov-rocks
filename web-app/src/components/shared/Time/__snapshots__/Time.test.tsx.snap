// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when 0 minutes passed 1`] = `""`;

exports[`matches snapshot when 1 minute passed 1`] = `
<span
  title="Time to Complete"
>
  1 minute
</span>
`;

exports[`matches snapshot when 12 minutes passed 1`] = `
<span
  title="Time to Complete"
>
  12 minutes
</span>
`;

exports[`matches snapshot when 60 minutes passed 1`] = `
<span
  title="Time to Complete"
>
  1 hour
</span>
`;

exports[`matches snapshot when 120 minutes passed 1`] = `
<span
  title="Time to Complete"
>
  2 hours
</span>
`;

exports[`matches snapshot when 121 minutes passed 1`] = `
<span
  title="Time to Complete"
>
  2 hours, 1 minute
</span>
`;

exports[`matches snapshot when 123 minutes passed 1`] = `
<span
  title="Time to Complete"
>
  2 hours, 3 minutes
</span>
`;
